tests/components/accuweather/test_sensor.py

```diff
@@ -0,0 +1,598 @@
+    assert entry
+    assert entry.unique_id == "0123456-uvindex"
+
+
+async def test_sensor_with_forecast(hass):
+    """Test states of the sensor with forecast."""
+    await init_integration(hass, forecast=True)
+    registry = await hass.helpers.entity_registry.async_get_registry()
+
+    state = hass.states.get("sensor.home_hours_of_sun_0d")
+    assert state
+    assert state.state == "7.2"
+    assert state.attributes.get(ATTR_ATTRIBUTION) == ATTRIBUTION
+    assert state.attributes.get(ATTR_ICON) == "mdi:weather-partly-cloudy"
+    assert state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) == TIME_HOURS
+
+    entry = registry.async_get("sensor.home_hours_of_sun_0d")
+    assert entry
+    assert entry.unique_id == "0123456-hoursofsun-0"
+
+    state = hass.states.get("sensor.home_realfeel_temperature_max_0d")
+    assert state
+    assert state.state == "29.8"
+    assert state.attributes.get(ATTR_ATTRIBUTION) == ATTRIBUTION
+    assert state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) == TEMP_CELSIUS
+    assert state.attributes.get(ATTR_DEVICE_CLASS) == DEVICE_CLASS_TEMPERATURE
+
+    entry = registry.async_get("sensor.home_realfeel_temperature_max_0d")
+    assert entry
+
+    state = hass.states.get("sensor.home_realfeel_temperature_min_0d")
+    assert state
+    assert state.state == "15.1"
+    assert state.attributes.get(ATTR_ATTRIBUTION) == ATTRIBUTION
+    assert state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) == TEMP_CELSIUS
+    assert state.attributes.get(ATTR_DEVICE_CLASS) == DEVICE_CLASS_TEMPERATURE
+
+    entry = registry.async_get("sensor.home_realfeel_temperature_min_0d")
+    assert entry
+    assert entry.unique_id == "0123456-realfeeltemperaturemin-0"
+
+    state = hass.states.get("sensor.home_thunderstorm_probability_day_0d")
+    assert state
+    assert state.state == "40"
+    assert state.attributes.get(ATTR_ATTRIBUTION) == ATTRIBUTION
+    assert state.attributes.get(ATTR_ICON) == "mdi:weather-lightning"
+    assert state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) == UNIT_PERCENTAGE
+
+    entry = registry.async_get("sensor.home_thunderstorm_probability_day_0d")
+    assert entry
+    assert entry.unique_id == "0123456-thunderstormprobabilityday-0"
+
+    state = hass.states.get("sensor.home_thunderstorm_probability_night_0d")
+    assert state
+    assert state.state == "40"
+    assert state.attributes.get(ATTR_ATTRIBUTION) == ATTRIBUTION
+    assert state.attributes.get(ATTR_ICON) == "mdi:weather-lightning"
+    assert state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) == UNIT_PERCENTAGE
+
+    entry = registry.async_get("sensor.home_thunderstorm_probability_night_0d")
+    assert entry
+    assert entry.unique_id == "0123456-thunderstormprobabilitynight-0"
+
+    state = hass.states.get("sensor.home_uv_index_0d")
+    assert state
+    assert state.state == "5"
+    assert state.attributes.get(ATTR_ATTRIBUTION) == ATTRIBUTION
+    assert state.attributes.get(ATTR_ICON) == "mdi:weather-sunny"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5985897)